js/utils/getBitcoinPrice.js

```diff
@@ -4,143 +4,164 @@ var __ = require('underscore'),
 
-  //some APIs require currency to be upper case
-  currency = currency.toUpperCase();
+    //some APIs require currency to be upper case
+    currency = currency.toUpperCase();
 
-  //if currency is Bitcoin, don't bother with finding the Bitcoin value
-  if(currency !== 'BTC')
-  {
+    //if currency is Bitcoin, don't bother with finding the Bitcoin value
+    if (currency !== 'BTC') {
 
-    var btPrices = [];
-
-    //initial call
-    //bitcoin average is hitting the rate limit
-    /*
-  $.ajax({
-    method: "GET",
-    url: "https://api.bitcoinaverage.com/ticker/global/" + currency
-  })
-      .done(function (response)
-      {
-        //console.log("bitcoinAverage: " + response['24h_avg']);
-        if($.isNumeric(response['24h_avg'])) {
-          btPrices.push(response['24h_avg']);
-        }
-      })
-      .fail(function (jqXHR, textStatus, errorThrown)
-      {
-        //console.log("bitcoinAverage request failed:");
-        //console.log(jqXHR);
-        //console.log(textStatus);
-        //console.log(errorThrown);
-      })
-      .always(function ()
-      {
-        callCoindesk();
-      });
-
-  var callCoindesk = function ()
-  {*/
-      $.ajax({
-        method: "GET",
-        dataType: "json",
-        url: "https://api.coindesk.com/v1/bpi/currentprice/" + currency + ".json"
-      })
-          .done(function (response)
-          {
-            //console.log("coinDesk: " + response.bpi[currency]['rate']);
-            if($.isNumeric(response.bpi[currency].rate)) {
-              btPrices.push(response.bpi[currency].rate);
-            }
-          })
-          .fail(function (jqXHR, textStatus, errorThrown)
-          {
-            //console.log("coinDesk request failed:");
-            //console.log(jqXHR);
-            //console.log(textStatus);
-            //console.log(errorThrown);
-          })
-          .always(function ()
-          {
-            callBlockchain();
-          });
-    //};
+        window.btcAverages = {};
+        var btPrices = [];
+        var btcAverages = {rates: {}};
 
+        var callBlockchain = function () {
+            $.ajax({
+                method: "GET",
+                url: "https://blockchain.info/ticker"
+            })
+                .done(function (response) {
+                    var BlockchainCurrencies = {};
+                    for (var bcCurrency in response) {
+                        if (response.hasOwnProperty(bcCurrency)) {
+                            BlockchainCurrencies[bcCurrency] = response[bcCurrency]['15m'];
+                        }
+                    }
+                    btPrices.push(BlockchainCurrencies);
+                })
+                .fail(function (jqXHR, textStatus, errorThrown) {
+                    console.log("Blockchain request failed: ");
+                    console.log(jqXHR);
+                    console.log(textStatus);
+                    console.log(errorThrown);
+                })
```

Comment:
mayb you want to put this logging pattern as a function, I see it being used in several places. #DRY